src/controller/java/CHIPDeviceController-JNI.cpp

```diff
@@ -385,6 +397,52 @@ JNI_METHOD(jlong, newDeviceController)(JNIEnv * env, jobject self, jobject contr
                 SuccessOrExit(err);
             }
         }
+
+        jobject countryCode;
+        err = chip::JniReferences::GetInstance().GetOptionalValue(countryCodeOptional, countryCode);
+        SuccessOrExit(err);
+
+        if (countryCode != nullptr)
+        {
+            jstring countryCodeStr = static_cast<jstring>(countryCode);
+            JniUtfString countryCodeJniString(env, countryCodeStr);
+
+            VerifyOrExit(countryCodeJniString.size() == 2, err = CHIP_ERROR_INVALID_ARGUMENT);
+
+            chip::Controller::CommissioningParameters commissioningParams = wrapper->GetCommissioningParameters();
+            commissioningParams.SetCountryCode(countryCodeJniString.charSpan());
+            err = wrapper->UpdateCommissioningParameters(commissioningParams);
```

Comment:
Please add a comment that indicates that `UpdateCommissioningParameters` has owning storage for the country code, which is why it's OK to just set a Span on SetCountryCode backed by an object that is transient